packages/grafana-data/src/types/datasource.ts

```diff
@@ -351,6 +351,19 @@ abstract class DataSourceApi<
    * @alpha -- experimental
    */
   getDefaultQuery?(app: CoreApp): Partial<TQuery>;
+
+  /*
+   * Optionally, use this method to track information about queries.
+   * Only executed once, during the dashboard initialization.
+   * @alpha -- experimental
+   */
+  trackQueries?(
```

Comment:
**Question:** is this something that we would like to allow for external plugins as well?

I am mostly asking this, because currently we are rejecting plugins that are trying to collect any kind of user usage metrics on their own. With this method we would kind of give an option to do it this API.

Reply:
Speaking for @grafana/cloud-provider-plugins  and @grafana/aws-plugins : we would definitely want to allow it for our external plugins as we aim to gather adoption info about all the data sources we own.

I suppose it would still be possible to reject plugins based on the implementation of this method?

Any reason why we reject them? Is that a security issue (collecting sensitive info)? Or feasibility (cross domain calls to their own services)?
I feel like any plugin developer would want to know which of their feature is most used... So having a way to get these insights is valuable. But I understand how it's not really doable within HG for instance for plugins not developed by grafana.

We could argue for now that since this method is @ alpha/experimental it should not be used by developers external to grafana?